core/math/vector2.h

```diff
@@ -129,6 +129,7 @@ struct [[nodiscard]] Vector2 {
 	Vector2 reflect(const Vector2 &p_normal) const;
 
 	bool is_equal_approx(const Vector2 &p_v) const;
+	bool is_semantic_equal(const Vector2 &p_v) const;
 	bool is_zero_approx() const;
```

Comment:
Given the syntax of the surrounding functions (both here and elsewhere), maybe `is_equal_semantic` would be more appropriate?

Reply:
I'm considering using `is_same` as I'd say it's more clear, only drawback is that if we expose these the `float` one in scripting won't be available, and you'd have to use `is_same` which is a dynamic method

Reply:
What `float` method are you referring to? The only `is_same` I'm aware of is the static method, which wouldn't overlap with this implementation.

Reply:
Meaning the [scripting method](https://docs.godotengine.org/en/latest/classes/class_%40globalscope.html#class-globalscope-method-is-same)

Reply:
I *imagine* that GDScript would have a means of binding a function to `float`; it'd be a bit unorthodox & probably more involved, but nothing outright inhibiting.

Reply:
We can't bing methods called on `float` currently, but I think `is_same` works using the generic version